gateway/server.go

```diff
@@ -491,11 +491,16 @@ func generateOAuthPrefix(apiID string) string {
 
 // Create API-specific OAuth handlers and respective auth servers
 func addOAuthHandlers(spec *APISpec, muxer *mux.Router) *OAuthManager {
-	apiAuthorizePath := spec.Proxy.ListenPath + "tyk/oauth/authorize-client{_:/?}"
-	clientAuthPath := spec.Proxy.ListenPath + "oauth/authorize{_:/?}"
-	clientAccessPath := spec.Proxy.ListenPath + "oauth/token{_:/?}"
-	revokeToken := spec.Proxy.ListenPath + "oauth/revoke"
-	revokeAllTokens := spec.Proxy.ListenPath + "oauth/revoke_all"
+	var pathSeparator string
+	if config.Global().SlaveOptions.BindToSlugsInsteadOfListenPaths {
+		pathSeparator = "/"
```

Comment:
I believe you should use Slug value instead of "/"

Reply:
Slug is already set in ListenPath:

```
if config.Global().SlaveOptions.BindToSlugsInsteadOfListenPaths {
   newListenPath := "/" + def.Slug //+ "/"
```